apps/src/StudioApp.js

```diff
@@ -152,6 +153,19 @@ var StudioApp = function () {
   this.maxRecommendedBlocksToFlag_ = 1;
 
   /**
+   * @typedef {Object} AuthoredHint
+   * @property {string} content
+   * @property {string} hintId
+   * @property {string} hintClass
+   * @property {string} hintType
+   * @property {boolean} alreadySeen
+   */
+  /**
+  * @type {!AuthoredHint[]}
+  */
+  this.authoredHints_ = [];
```

Comment:
I might be wrong, but it looks like very little of the logic you've added to StudioApp.js actually depends on `StudioApp`.  I see you've already broken a lot out into authoredHintUtils.  Since this is one of our files that's already over 2K lines, I'd love to see even more of this logic pulled out into a helper class (e.g. `AuthoredHintEngine`) that StudioApp makes an instance of on construction and uses as needed - just for the sake of organization, really.  Others might disagree :grin:.